### DISCUSSION TITLE
Allow customisation of upstream image timeout

### DISCUSSION THREAD

**Discussion by bill-zhanxg** [vercel/next.js#72951] [Ideas]
### Goals

Allow loading and optimisation of extremely large image and not just a hard coded 7 seconds

### Background

I'm trying to load extremely large image (20GB) but it throws `DOMException {stack: 'TimeoutError: The operation was aborted due t…ss.processTimers (node:internal/timers:529:7)'}` after 7 seconds

#65821

### Proposal

Add an experimental config in next.config.ts for external image fetching just like the imgOptTimeoutInSeconds option for sharp timeout.

https://github.com/vercel/next.js/blob/720fe3577eb58050966a44bc1fa88e01fcac3962/packages/next/src/server/image-optimizer.ts#L568

**Comment by arashi-dev** [comment]
I also need this!

**Comment by kyoung-jnn** [comment]
Is there a specific reason why it was hardcoded to 7 seconds instead of another value?

**Reply by anay-208** [reply]
#65821

**Comment by siamahnaf** [comment]
Anyone find any solutions for this?

**Reply by bill-zhanxg** [reply]
I think the only solution is to fork Nextjs and modify the source code yourself sadly.

**Reply by siamahnaf** [reply]
Then why you are waiting for. Fork today and spread it world wide.

**Reply by bill-zhanxg** [reply]
I currently don't need this feature rn

**Comment by sahanatvessel** [comment]
@leerob any insight on this please?

**Comment by ShayokhShorfuddin** [comment]
Need this feature badly

**Comment by Akhilez** [comment]
I'm using `<img/>` for now, but would like to see some custom timeout

**Comment by dislick** [comment]
I'm currently at my in-laws during the holidays and their Wi-Fi is terrible. I keep running into this issue, even for regularly sized images. A customizable timeout would be great.
